lib/ansible/module_utils/network/iosxr/facts/lacp_interfaces/lacp_interfaces.py

```diff
@@ -0,0 +1,104 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The iosxr lacp_interfaces fact class
+It is in this file the configuration is collected from the device
+for a given resource, parsed, and the facts tree is populated
+based on the configuration.
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+import re
+from copy import deepcopy
+
+from ansible.module_utils.network.common import utils
+from ansible.module_utils.network.iosxr.argspec.lacp_interfaces.lacp_interfaces import Lacp_interfacesArgs
+from ansible.module_utils.six import iteritems
+
+
+class Lacp_interfacesFacts(object):
+    """ The iosxr lacp_interfaces fact class
+    """
+
+    def __init__(self, module, subspec='config', options='options'):
+        self._module = module
+        self.argument_spec = Lacp_interfacesArgs.argument_spec
+        spec = deepcopy(self.argument_spec)
+        if subspec:
+            if options:
+                facts_argument_spec = spec[subspec][options]
+            else:
+                facts_argument_spec = spec[subspec]
+        else:
+            facts_argument_spec = spec
+
+        self.generated_spec = utils.generate_dict(facts_argument_spec)
+
+    def populate_facts(self, connection, ansible_facts, data=None):
+        """ Populate the facts for lacp_interfaces
+        :param connection: the device connection
+        :param ansible_facts: Facts dictionary
+        :param data: previously collected conf
+        :rtype: dictionary
+        :returns: facts
+        """
+
+        if not data:
+            data = connection.get_config(flags='interface')
+            interfaces = data.split('interface ')
+
+        objs = []
+        for interface in interfaces:
+            obj = self.render_config(self.generated_spec, interface)
+            if obj:
+                objs.append(obj)
+
+        ansible_facts['ansible_network_resources'].pop('lacp_interfaces', None)
+        facts = {}
+        if objs:
+            facts['lacp_interfaces'] = []
+            params = utils.validate_config(self.argument_spec, {'config': objs})
+            for cfg in params['config']:
+                facts['lacp_interfaces'].append(utils.remove_empties(cfg))
+
+        ansible_facts['ansible_network_resources'].update(facts)
+        return ansible_facts
+
+    def render_config(self, spec, conf):
+        """
+        Render config as dictionary structure and delete keys
+          from spec for null values
+
+        :param spec: The facts tree, generated from the argspec
+        :param conf: The configuration
+        :rtype: dictionary
+        :returns: The generated config
+        """
+        config = deepcopy(spec)
+
+        match = re.search(r'(GigabitEthernet|Bundle-Ether|TenGigE|FortyGigE|HundredGigE)(\S+)', conf, re.M)
```

Comment:
I think this can be handled via `normalize_interface` fn., like it was done before in generic modules as well (ref: https://github.com/ansible/ansible/pull/59853/files#diff-1be88b42184b2d05a46b6b97ee34a92eR85). Once respective PR is merged u can utilise the fn. and remove the regex search.

Reply:
@justjais `normalize_interface` seems to convert interface names to proper structure - i.e., if user passes `gigE0/0/0/1` in the task, it would convert it to `GigabitEthernet0/0/0/1`. This seems redundant while gathering fact because the interface names will always be in the proper structure when fetched from the device. Not sure how it would replace the regex search. Also, we need this regex in order to fetch only those interfaces that support lacp configurations.

Reply:
So, here your regex search takes an input e.g. `interface GigabitEthernet0/0/0/1` and matches with regex and then sets the `config['name'] = GigabitEthernet0/0/0/1` which is what `normalize_interface` does, just that you'll have to add `Bundle-Ether` entry, coz currently it's not there.

Reply:
@justjais - The reason to use the regex is to fetch only those interfaces that support lacp configuration and leave out unnecessary ones for this module.

Reply:
@justjais `normalize_interfaces` is aimed at sanitizing user input, and seems to be redundant while parsing running-config where the interface names are already normalized.